src/PythonWrapper.cpp

```diff
@@ -30,24 +31,37 @@ using namespace std;
 const char *VERSION =
         "PYTHON_CLIENT_VERSION: " PYTHON_CLIENT_VERSION ", BUILD DATE: " PYCLI_BUILD_DATE " ";
 
-map<CPushConsumer *, PyObject *> g_CallBackMap;
+map<CPushConsumer *, pair<PyObject *, object>> g_CallBackMap;
 
 class PyThreadStateLock {
 public:
-    PyThreadStateLock(void) {
+    PyThreadStateLock() {
         state = PyGILState_Ensure();
     }
 
-    ~PyThreadStateLock(void) {
-        if (state == PyGILState_LOCKED) {
-            PyGILState_Release(state);
-        }
+    ~PyThreadStateLock() {
+        // NOTE: 必须跟 PyGILState_Ensure 成对出现，否则可能出现死锁!!!
+        PyGILState_Release(state);
```

Comment:
Sometimes, the state of the GIL is not PyGILState_LOCKED,  and the function PyGILState_Release will crash. Did you find this before？

Reply:
the condition statement cause deadlock in my code. so i remove it.

Reply:
i want run it in my macbook for debug other bugs, i am doing the work now. i will commit a new PR after finish.